### DISCUSSION TITLE
GitRepository displaying error  failed to configure checkout strategy for Git implementation '': unsupported Git implementation ''

### DISCUSSION THREAD

**Discussion by PutuGandi** [fluxcd/flux2#4190] [General]
Hai all, i have issue with flux 0.38.3 , the GitRepository suddenly display message like this:
` failed to configure checkout strategy for Git implementation '': unsupported Git implementation ''`
 
Did anyone have any problems with this?
 
 Thank You

**Comment by makkes** [comment]
Are you sure you're running 0.38.3? This message has been removed way before that release. Please paste the output of `flux check` here.

**Reply by PutuGandi** [reply]
hi @makkes , this is the output: 

**flux check** : 
```
 flux 0.38.3 <2.1.0 (new version is available, please upgrade)
✔ Kubernetes 1.22.17-eks-2d98532 >=1.20.6-0
► checking controllers
✔ helm-controller: deployment ready
► ghcr.io/fluxcd/helm-controller:v0.21.0
✔ kustomize-controller: deployment ready
► ghcr.io/fluxcd/kustomize-controller:v0.25.0
✔ notification-controller: deployment ready
► ghcr.io/fluxcd/notification-controller:v0.23.5
✔ source-controller: deployment ready
► ghcr.io/fluxcd/source-controller:v0.24.4
► checking crds
✔ alerts.notification.toolkit.fluxcd.io/v1beta2
✔ buckets.source.toolkit.fluxcd.io/v1beta2
✔ gitrepositories.source.toolkit.fluxcd.io/v1
✔ helmcharts.source.toolkit.fluxcd.io/v1beta2
✔ helmreleases.helm.toolkit.fluxcd.io/v2beta1
✔ helmrepositories.source.toolkit.fluxcd.io/v1beta2
✔ kustomizations.kustomize.toolkit.fluxcd.io/v1
✔ ocirepositories.source.toolkit.fluxcd.io/v1beta2
✔ providers.notification.toolkit.fluxcd.io/v1beta2
✔ receivers.notification.toolkit.fluxcd.io/v1
✔ all checks passed
```

**flux version:** 
`flux version 0.38.3`

**Reply by makkes** [reply]
Your cluster is running source-controller v0.24.4 which is a much older version than the one that Flux 0.38.3 [ships with](https://github.com/fluxcd/flux2/blob/v0.38.3/manifests/bases/source-controller/kustomization.yaml#L4). Please make sure to upgrade all Flux components installed on your cluster using [the upgrade guide](https://fluxcd.io/flux/installation/upgrade/) to align with 0.38.3. Then try again.

**Reply by PutuGandi** [reply]
hi @makkes  

if i try upgrade flux with this command `https://fluxcd.io/flux/installation/upgrade/#upgrade-with-flux-cli` , will it be get impact to the resource ? i mean like recreate resources , because im affraid if Load Balancer that created by Ingress will be recreate.

**Reply by makkes** [reply]
No, resources will not be re-created if they haven't changed in Git.

**Reply by PutuGandi** [reply]
thank you @makkes
